File: ios/EnterpriseMobileApp/Sources/Core/Domain/Repositories/AuthRepository.swift
import Foundation

/// Repository protocol for authentication operations.
public protocol AuthRepository: Sendable {
    func login(email: String, password: String) async -> DomainResult<User>
    func loginWithBiometric() async -> DomainResult<User>
    func loginWithPasskey(challenge: Data) async -> DomainResult<User>
    func logout() async -> DomainResult<Void>
    func getCurrentUser() async -> DomainResult<User>
    func refreshToken() async -> DomainResult<Void>
}

File: android/core/domain/src/main/java/com/enterprise/core/domain/repositories/AuthRepository.kt
package com.enterprise.core.domain.repositories

import com.enterprise.core.domain.entities.DomainResult
import com.enterprise.core.domain.entities.User

/** Authentication repository contract - no OS/framework dependencies. */
interface AuthRepository {
    suspend fun login(email: String, password: String): DomainResult<User>
    suspend fun loginWithBiometric(): DomainResult<User>
    suspend fun loginWithPasskey(challenge: ByteArray): DomainResult<User>
    suspend fun logout(): DomainResult<Unit>
    suspend fun getCurrentUser(): DomainResult<User>
    suspend fun refreshToken(): DomainResult<Unit>
}
